Train Model YOLO

In [3]:
!pip install ultralytics

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 25.6 MB/s eta 0:00:0000:01


Collecting ultralytics
  Downloading ultralytics-8.4.41-py3-none-any.whl.metadata (39 kB)
Requirement already satisfied: numpy>=1.23.0 in /usr/local/lib/python3.12/dist-packages (from ultralytics) (2.0.2)
Requirement already satisfied: matplotlib>=3.3.0 in /usr/local/lib/python3.12/dist-packages (from ultralytics) (3.10.0)
Requirement already satisfied: opencv-python>=4.6.0 in /usr/local/lib/python3.12/dist-packages (from ultralytics) (4.13.0.92)
Requirement already satisfied: pillow>=7.1.2 in /usr/local/lib/python3.12/dist-packages (from ultralytics) (11.3.0)
Requirement already satisfied: pyyaml>=5.3.1 in /usr/local/lib/python3.12/dist-packages (from ultralytics) (6.0.3)
Requirement already satisfied: requests>=2.23.0 in /usr/local/lib/python3.12/dist-packages (from ultralytics) (2.32.4)
Requirement already satisfied: scipy>=1.4.1 in /usr/local/lib/python3.12/dist-packages (from ultralytics) (1.16.3)
Requirement already satisfied: torch>=1.8.0 in /usr/local/lib/python3.12/dist-packages (from ultralytics) (2.10.0+cu128)
Requirement already satisfied: torchvision>=0.9.0 in /usr/local/lib/python3.12/dist-packages (from ultralytics) (0.25.0+cu128)
Requirement already satisfied: psutil>=5.8.0 in /usr/local/lib/python3.12/dist-packages (from ultralytics) (5.9.5)
Requirement already satisfied: polars>=0.20.0 in /usr/local/lib/python3.12/dist-packages (from ultralytics) (1.35.2)
Collecting ultralytics-thop>=2.0.18 (from ultralytics)
  Downloading ultralytics_thop-2.0.18-py3-none-any.whl.metadata (14 kB)
Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.12/dist-packages (from matplotlib>=3.3.0->ultralytics) (1.3.3)
Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.12/dist-packages (from matplotlib>=3.3.0->ultralytics) (0.12.1)
Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.12/dist-packages (from matplotlib>=3.3.0->ultralytics) (4.61.1)
Requirement already satisfied: kiwisolver>=1.3.1 in /usr/local/lib/python3.12/dist-packages (from matplotlib>=3.3.0->ultralytics) (1.4.9)
Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.12/dist-packages (from matplotlib>=3.3.0->ultralytics) (26.0)
Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.12/dist-packages (from matplotlib>=3.3.0->ultralytics) (3.3.2)
Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.12/dist-packages (from matplotlib>=3.3.0->ultralytics) (2.9.0.post0)
Requirement already satisfied: polars-runtime-32==1.35.2 in /usr/local/lib/python3.12/dist-packages (from polars>=0.20.0->ultralytics) (1.35.2)
Requirement already satisfied: charset_normalizer<4,>=2 in /usr/local/lib/python3.12/dist-packages (from requests>=2.23.0->ultralytics) (3.4.4)
Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.12/dist-packages (from requests>=2.23.0->ultralytics) (3.11)
Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.12/dist-packages (from requests>=2.23.0->ultralytics) (2.5.0)
Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.12/dist-packages (from requests>=2.23.0->ultralytics) (2026.1.4)
Requirement already satisfied: filelock in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (3.24.3)
Requirement already satisfied: typing-extensions>=4.10.0 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (4.15.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (75.2.0)
Requirement already satisfied: sympy>=1.13.3 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (1.14.0)
Requirement already satisfied: networkx>=2.5.1 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (3.6.1)
Requirement already satisfied: jinja2 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (3.1.6)
Requirement already satisfied: fsspec>=0.8.5 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (2026.2.0)
Requirement already satisfied: cuda-bindings==12.9.4 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (12.9.4)
Requirement already satisfied: nvidia-cuda-nvrtc-cu12==12.8.93 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (12.8.93)
Requirement already satisfied: nvidia-cuda-runtime-cu12==12.8.90 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (12.8.90)
Requirement already satisfied: nvidia-cuda-cupti-cu12==12.8.90 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (12.8.90)
Requirement already satisfied: nvidia-cudnn-cu12==9.10.2.21 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (9.10.2.21)
Requirement already satisfied: nvidia-cublas-cu12==12.8.4.1 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (12.8.4.1)
Requirement already satisfied: nvidia-cufft-cu12==11.3.3.83 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (11.3.3.83)
Requirement already satisfied: nvidia-curand-cu12==10.3.9.90 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (10.3.9.90)
Requirement already satisfied: nvidia-cusolver-cu12==11.7.3.90 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (11.7.3.90)
Requirement already satisfied: nvidia-cusparse-cu12==12.5.8.93 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (12.5.8.93)
Requirement already satisfied: nvidia-cusparselt-cu12==0.7.1 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (0.7.1)
Requirement already satisfied: nvidia-nccl-cu12==2.27.5 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (2.27.5)
Requirement already satisfied: nvidia-nvshmem-cu12==3.4.5 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (3.4.5)
Requirement already satisfied: nvidia-nvtx-cu12==12.8.90 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (12.8.90)
Requirement already satisfied: nvidia-nvjitlink-cu12==12.8.93 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (12.8.93)
Requirement already satisfied: nvidia-cufile-cu12==1.13.1.3 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (1.13.1.3)
Requirement already satisfied: triton==3.6.0 in /usr/local/lib/python3.12/dist-packages (from torch>=1.8.0->ultralytics) (3.6.0)
Requirement already satisfied: cuda-pathfinder~=1.1 in /usr/local/lib/python3.12/dist-packages (from cuda-bindings==12.9.4->torch>=1.8.0->ultralytics) (1.3.5)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.12/dist-packages (from python-dateutil>=2.7->matplotlib>=3.3.0->ultralytics) (1.17.0)
Requirement already satisfied: mpmath<1.4,>=1.1.0 in /usr/local/lib/python3.12/dist-packages (from sympy>=1.13.3->torch>=1.8.0->ultralytics) (1.3.0)
Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.12/dist-packages (from jinja2->torch>=1.8.0->ultralytics) (3.0.3)
Downloading ultralytics-8.4.41-py3-none-any.whl (1.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 25.6 MB/s eta 0:00:0000:01
Downloading ultralytics_thop-2.0.18-py3-none-any.whl (28 kB)
Installing collected packages: ultralytics-thop, ultralytics
Successfully installed ultralytics-8.4.41 ultralytics-thop-2.0.18

In [4]:
%%writefile data.yml
path: /kaggle/input/datasets/bindang/license-plate-dataset/data_bs1

train: images/train
val: images/val

nc: 1
names: ['plate']

Overwriting data.yml


Overwriting data.yml

In [5]:
from ultralytics import YOLO

model = YOLO("yolov8s.pt")

results = model.train(data="data.yml", 
                      epochs=100, 
                      imgsz=960, 
                      degrees=10,
                      scale=0.5,
                      fliplr=0.5,
                      batch=16)

Creating new Ultralytics Settings v0.0.6 file ✅ 
View Ultralytics Settings with 'yolo settings' or at '/root/.config/Ultralytics/settings.json'
Update Settings with 'yolo settings key=value', i.e. 'yolo settings runs_dir=path/to/dir'. For help see https://docs.ultralytics.com/quickstart/#ultralytics-settings.
Ultralytics 8.4.41 🚀 Python-3.12.12 torch-2.10.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
engine/trainer: agnostic_nms=False, amp=True, angle=1.0, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, cls_pw=0.0, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=data.yml, degrees=10, deterministic=True, device=None, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, end2end=None, epochs=100, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=960, int8=False,

Creating new Ultralytics Settings v0.0.6 file ✅
View Ultralytics Settings with 'yolo settings' or at '/root/.config/Ultralytics/settings.json'
Update Settings with 'yolo settings key=value', i.e. 'yolo settings runs_dir=path/to/dir'. For help see https://docs.ultralytics.com/quickstart/#ultralytics-settings.
Downloading https://github.com/ultralytics/assets/releases/download/v8.4.0/yolov8s.pt to 'yolov8s.pt': 100% ━━━━━━━━━━━━ 21.5MB 219.5MB/s 0.1s
Ultralytics 8.4.41 🚀 Python-3.12.12 torch-2.10.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
engine/trainer: agnostic_nms=False, amp=True, angle=1.0, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, cls_pw=0.0, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=data.yml, degrees=10, deterministic=True, device=None, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, end2end=None, epochs=100, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=960, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolov8s.pt, momentum=0.937, mosaic=1.0, multi_scale=0.0, name=train, nbs=64, nms=False, opset=None, optimize=False, optimizer=auto, overlap_mask=True, patience=100, perspective=0.0, plots=True, pose=12.0, pretrained=True, profile=False, project=None, rect=False, resume=False, retina_masks=False, rle=1.0, save=True, save_conf=False, save_crop=False, save_dir=/kaggle/working/runs/detect/train, save_frames=False, save_json=False, save_period=-1, save_txt=False, scale=0.5, seed=0, shear=0.0, show=False, show_boxes=True, show_conf=True, show_labels=True, simplify=True, single_cls=False, source=None, split=val, stream_buffer=False, task=detect, time=None, tracker=botsort.yaml, translate=0.1, val=True, verbose=True, vid_stride=1, visualize=False, warmup_bias_lr=0.1, warmup_epochs=3.0, warmup_momentum=0.8, weight_decay=0.0005, workers=8, workspace=None
Downloading https://ultralytics.com/assets/Arial.ttf to '/root/.config/Ultralytics/Arial.ttf': 100% ━━━━━━━━━━━━ 755.1KB 23.9MB/s 0.0s
Overriding model.yaml nc=80 with nc=1

                   from  n    params  module                                       arguments
  0                  -1  1       928  ultralytics.nn.modules.conv.Conv             [3, 32, 3, 2]
  1                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]
  2                  -1  1     29056  ultralytics.nn.modules.block.C2f             [64, 64, 1, True]
  3                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]
  4                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]
  5                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]
  6                  -1  2    788480  ultralytics.nn.modules.block.C2f             [256, 256, 2, True]
  7                  -1  1   1180672  ultralytics.nn.modules.conv.Conv             [256, 512, 3, 2]
  8                  -1  1   1838080  ultralytics.nn.modules.block.C2f             [512, 512, 1, True]
  9                  -1  1    656896  ultralytics.nn.modules.block.SPPF            [512, 512, 5]
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 12                  -1  1    591360  ultralytics.nn.modules.block.C2f             [768, 256, 1]
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 15                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]
 16                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 18                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]
 19                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 21                  -1  1   1969152  ultralytics.nn.modules.block.C2f             [768, 512, 1]
 22        [15, 18, 21]  1   2116435  ultralytics.nn.modules.head.Detect           [1, 16, None, [128, 256, 512]]
Model summary: 130 layers, 11,135,987 parameters, 11,135,971 gradients, 28.6 GFLOPs

Transferred 349/355 items from pretrained weights
Freezing layer 'model.22.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks...
Downloading https://github.com/ultralytics/assets/releases/download/v8.4.0/yolo26n.pt to 'yolo26n.pt': 100% ━━━━━━━━━━━━ 5.3MB 104.6MB/s 0.1s
AMP: checks passed ✅
train: Fast image access ✅ (ping: 0.4±0.2 ms, read: 9.5±3.6 MB/s, size: 31.0 KB)
train: Scanning /kaggle/input/datasets/bindang/license-plate-dataset/data_bs1/labels/train... 6606 images, 0 backgrounds, 1 corrupt: 100% ━━━━━━━━━━━━ 6607/6607 301.2it/s 21.9s<0.0s
train: /kaggle/input/datasets/bindang/license-plate-dataset/data_bs1/images/train/quandoi63.jpg: ignoring corrupt image/label: [Errno 30] Read-only file system: '/kaggle/input/datasets/bindang/license-plate-dataset/data_bs1/images/train/quandoi63.jpg'
WARNING ⚠️ train: Cache directory /kaggle/input/datasets/bindang/license-plate-dataset/data_bs1/labels is not writable, cache not saved.
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 8.1±5.1 MB/s, size: 32.9 KB)
val: Scanning /kaggle/input/datasets/bindang/license-plate-dataset/data_bs1/labels/val... 1652 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 1652/1652 310.9it/s 5.3s0.1s
WARNING ⚠️ val: Cache directory /kaggle/input/datasets/bindang/license-plate-dataset/data_bs1/labels is not writable, cache not saved.
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically...
optimizer: MuSGD(lr=0.01, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias(decay=0.0)
Plotting labels to /kaggle/working/runs/detect/train/labels.jpg...
Image sizes 960 train, 960 val
Using 2 dataloader workers
Logging results to /kaggle/working/runs/detect/train
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      1/100      7.93G      1.287      1.263       1.27         23        960: 100% ━━━━━━━━━━━━ 413/413 1.6it/s 4:22<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.0it/s 17.3s0.3s
                   all       1652       1667      0.967      0.978      0.986       0.65

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      2/100      7.27G      1.192     0.6202       1.19         29        960: 100% ━━━━━━━━━━━━ 413/413 1.6it/s 4:10<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.975      0.974      0.985      0.546

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      3/100      7.27G      1.212     0.6709      1.214         22        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:04<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.1it/s 16.6s0.3s
                   all       1652       1667      0.956      0.963      0.986      0.627

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      4/100      7.28G      1.233     0.6884      1.241         27        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.1it/s 16.7s0.3s
                   all       1652       1667      0.976      0.945      0.973      0.638

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      5/100      7.28G      1.193     0.6458      1.224         28        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.1it/s 16.6s0.3s
                   all       1652       1667       0.98      0.978      0.991      0.613

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      6/100      7.28G      1.166      0.614      1.217         32        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.1it/s 16.6s0.3s
                   all       1652       1667      0.979      0.985      0.992      0.569

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      7/100      7.28G      1.153     0.5832      1.216         28        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.985      0.984      0.993      0.662

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      8/100      7.95G      1.155      0.573      1.215         27        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.987       0.98      0.993      0.631

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      9/100      7.28G      1.116     0.5473      1.201         31        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:00<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667       0.99      0.982      0.994      0.633

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     10/100      7.28G      1.124     0.5444      1.206         29        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.992      0.981      0.994       0.59

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     11/100      7.28G      1.095     0.5284      1.185         23        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.985       0.99      0.993      0.667

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     12/100      7.28G       1.09     0.5203      1.189         27        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.993      0.988      0.994       0.66

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     13/100      7.28G      1.084     0.5076      1.182         29        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667       0.99       0.99      0.994      0.702

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     14/100      7.28G      1.073     0.5053      1.178         29        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.991      0.992      0.993      0.698

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     15/100      7.98G      1.063     0.4897       1.18         24        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.993      0.984      0.994      0.692

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     16/100      7.24G      1.063     0.4832      1.171         25        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.2s0.3s
                   all       1652       1667      0.984      0.992      0.994      0.689

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     17/100      7.24G      1.065     0.4823       1.18         26        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.984      0.993      0.993       0.68

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     18/100      7.24G      1.057     0.4797      1.178         38        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.989      0.989      0.994      0.706

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     19/100      7.31G      1.049     0.4746      1.166         22        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.1it/s 16.5s0.3s
                   all       1652       1667      0.992      0.988      0.994       0.69

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     20/100      7.24G      1.055     0.4775      1.168         28        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.992       0.99      0.994      0.684

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     21/100      7.24G      1.037     0.4633      1.167         31        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.992      0.993      0.994      0.678

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     22/100      7.24G      1.041      0.461      1.168         20        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.994      0.995      0.994      0.695

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     23/100      7.24G      1.028     0.4533       1.16         32        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.997      0.994      0.994      0.692

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     24/100      7.24G      1.027     0.4543      1.163         31        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.987      0.992      0.993      0.699

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     25/100      7.24G      1.024     0.4497      1.161         38        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.994      0.992      0.994      0.696

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     26/100      7.24G      1.031     0.4528      1.164         30        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.2s0.3s
                   all       1652       1667      0.994      0.995      0.995      0.708

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     27/100      7.24G      1.018     0.4409      1.153         29        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.992      0.996      0.994       0.69

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     28/100      7.24G      1.012     0.4395      1.147         30        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.992      0.994      0.993      0.704

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     29/100      7.24G      1.014     0.4408      1.158         22        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.996      0.993      0.994      0.694

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     30/100      7.24G      1.013      0.442      1.141         35        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.996      0.996      0.994      0.708

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     31/100      7.24G      1.014     0.4353      1.143         27        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.1it/s 16.5s0.3s
                   all       1652       1667      0.994      0.996      0.994      0.705

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     32/100      7.24G     0.9973     0.4335      1.136         35        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.995      0.997      0.994       0.71

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     33/100      7.24G      1.001     0.4315      1.147         27        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.1it/s 16.5s0.3s
                   all       1652       1667      0.996      0.996      0.994      0.708

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     34/100      7.24G     0.9946     0.4234      1.137         25        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.994      0.992      0.994      0.712

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     35/100      7.24G     0.9923     0.4224       1.14         24        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.996      0.995      0.995      0.714

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     36/100      7.24G     0.9885     0.4216      1.142         29        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.997      0.996      0.994      0.711

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     37/100      7.24G     0.9858     0.4229      1.139         31        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.998      0.998      0.994      0.711

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     38/100      7.24G     0.9803     0.4153      1.132         27        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.998      0.996      0.993      0.709

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     39/100      7.24G     0.9773     0.4094      1.138         23        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.998      0.996      0.994      0.716

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     40/100      7.24G     0.9742     0.4111      1.128         25        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.993      0.993      0.994      0.701

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     41/100      7.24G     0.9823      0.413      1.131         31        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.993      0.995      0.994      0.706

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     42/100      7.24G     0.9771     0.4062      1.137         34        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.995      0.996      0.994      0.716

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     43/100      7.24G     0.9748     0.4058      1.138         37        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.995      0.995      0.994      0.714

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     44/100      7.24G     0.9699     0.4056      1.132         29        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.2s0.3s
                   all       1652       1667      0.994      0.997      0.994       0.72

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     45/100      7.24G     0.9673      0.405      1.129         32        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.995      0.996      0.994       0.72

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     46/100      7.24G     0.9544      0.397      1.123         35        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.995      0.998      0.994       0.72

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     47/100      7.24G     0.9538      0.394      1.128         25        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.997      0.997      0.994       0.72

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     48/100      7.24G     0.9629     0.3929      1.129         30        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.996      0.997      0.994      0.717

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     49/100      7.24G      0.958     0.3929      1.125         22        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.997      0.995      0.994      0.721

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     50/100      7.24G     0.9535     0.3933      1.118         25        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.1it/s 16.6s0.3s
                   all       1652       1667      0.996      0.995      0.994      0.722

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     51/100      7.24G     0.9514     0.3903      1.124         27        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.995      0.993      0.994       0.72

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     52/100      7.24G     0.9449       0.39      1.125         24        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.996      0.993      0.994      0.717

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     53/100      7.24G     0.9433      0.388       1.11         21        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:03<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.1it/s 16.6s0.3s
                   all       1652       1667      0.998      0.995      0.995      0.721

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     54/100      7.24G     0.9439     0.3846      1.114         31        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.993      0.995      0.994      0.719

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     55/100      7.24G     0.9384     0.3849      1.116         33        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.996      0.996      0.995      0.719

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     56/100      7.24G     0.9358     0.3788      1.112         24        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:03<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.997      0.996      0.994      0.726

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     57/100      7.24G     0.9397     0.3793      1.115         34        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.996      0.996      0.994      0.719

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     58/100      7.24G     0.9376     0.3812      1.109         25        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.997      0.996      0.994      0.725

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     59/100      7.24G     0.9247     0.3738      1.099         24        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.998      0.996      0.995      0.725

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     60/100      7.24G     0.9174     0.3761      1.106         24        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.995      0.996      0.994      0.723

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     61/100      7.24G     0.9208     0.3742      1.107         33        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.995      0.999      0.994      0.727

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     62/100      7.24G     0.9181      0.374       1.11         31        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.1it/s 16.5s0.3s
                   all       1652       1667      0.997      0.997      0.995      0.722

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     63/100      7.24G     0.9237     0.3767      1.101         32        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:03<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.996      0.997      0.995      0.722

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     64/100      7.24G      0.913     0.3683      1.098         28        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.997      0.996      0.994      0.726

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     65/100      7.24G      0.912     0.3728      1.097         26        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.998      0.995      0.994      0.725

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     66/100      7.24G     0.8973     0.3621      1.103         28        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.998      0.997      0.994      0.722

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     67/100      7.24G     0.9072      0.364      1.096         32        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.996      0.995      0.994      0.727

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     68/100      7.24G     0.9063     0.3628      1.099         31        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.996      0.995      0.994      0.724

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     69/100      7.24G     0.8982       0.36      1.098         30        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:03<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.996      0.996      0.994      0.723

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     70/100      7.24G     0.9076     0.3596      1.095         23        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.997      0.997      0.994      0.725

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     71/100      7.24G     0.8954     0.3569      1.095         21        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.996      0.997      0.994      0.726

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     72/100      7.24G     0.8923     0.3527      1.091         28        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.997      0.997      0.995      0.725

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     73/100      7.24G     0.8913     0.3522      1.087         26        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.997      0.996      0.995      0.725

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     74/100      7.24G     0.8844     0.3483      1.088         24        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.995      0.996      0.994      0.725

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     75/100      7.24G     0.8911     0.3515      1.089         23        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.996      0.996      0.994      0.726

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     76/100      7.24G     0.8762      0.349      1.086         21        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:03<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.996      0.996      0.994      0.723

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     77/100      7.24G     0.8841     0.3471      1.088         30        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:03<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.996      0.995      0.995      0.723

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     78/100      7.24G     0.8734     0.3412      1.081         28        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:03<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.997      0.995      0.994      0.724

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     79/100      7.24G     0.8717     0.3428      1.078         31        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:03<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.996      0.996      0.994      0.725

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     80/100      7.24G     0.8707     0.3408       1.08         24        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:03<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.995      0.996      0.995      0.726

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     81/100      7.24G     0.8621     0.3388      1.074         30        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.996      0.996      0.994      0.724

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     82/100      7.24G     0.8707     0.3402      1.084         27        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:03<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.996      0.996      0.994      0.722

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     83/100      7.24G     0.8586     0.3344      1.076         30        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:03<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.996      0.997      0.994      0.721

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     84/100      7.24G     0.8548     0.3329      1.072         19        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.996      0.997      0.994      0.724

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     85/100      7.24G     0.8518     0.3304      1.073         28        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:03<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.995      0.996      0.994      0.724

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     86/100      7.24G     0.8433      0.329      1.064         22        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:03<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.995      0.997      0.995      0.723

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     87/100      7.24G     0.8514     0.3314      1.072         30        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:03<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.996      0.996      0.995      0.724

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     88/100      7.24G     0.8384      0.324      1.062         24        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.996      0.995      0.995      0.724

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     89/100      7.24G      0.835     0.3225      1.068         26        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:03<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.998      0.995      0.995      0.725

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     90/100      7.24G     0.8291      0.318      1.062         31        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.997      0.996      0.994      0.724
Closing dataloader mosaic
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     91/100      7.24G     0.8124     0.2859      1.083         14        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.995      0.998      0.995      0.724

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     92/100      7.24G     0.8064     0.2837      1.082         14        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.995      0.997      0.995      0.724

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     93/100      7.24G     0.8007     0.2835      1.077         14        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.995      0.997      0.995      0.721

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     94/100      7.24G     0.7907      0.279      1.074         14        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.3s0.3s
                   all       1652       1667      0.998      0.994      0.994      0.723

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     95/100      7.24G     0.7879     0.2778      1.075         14        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.998      0.995      0.994       0.72

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     96/100      7.24G     0.7824     0.2756      1.075         14        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.997      0.995      0.995      0.723

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     97/100      7.24G     0.7843     0.2752      1.071         14        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.995      0.997      0.995      0.722

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     98/100      7.24G     0.7776     0.2717      1.066         14        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:02<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.4s0.3s
                   all       1652       1667      0.997      0.995      0.995      0.724

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     99/100      7.24G     0.7722     0.2693      1.065         15        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.6s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.1it/s 16.6s0.3s
                   all       1652       1667      0.996      0.996      0.995      0.723

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
    100/100      7.24G     0.7718     0.2691      1.065         15        960: 100% ━━━━━━━━━━━━ 413/413 1.7it/s 4:01<0.5s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 3.2it/s 16.5s0.3s
                   all       1652       1667      0.997      0.995      0.995      0.722

100 epochs completed in 7.191 hours.
Optimizer stripped from /kaggle/working/runs/detect/train/weights/last.pt, 22.6MB
Optimizer stripped from /kaggle/working/runs/detect/train/weights/best.pt, 22.6MB

Validating /kaggle/working/runs/detect/train/weights/best.pt...
Ultralytics 8.4.41 🚀 Python-3.12.12 torch-2.10.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
Model summary (fused): 73 layers, 11,125,971 parameters, 0 gradients, 28.4 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 52/52 2.9it/s 17.8s0.3s
                   all       1652       1667      0.995      0.999      0.994      0.727
Speed: 0.3ms preprocess, 6.7ms inference, 0.0ms loss, 0.9ms postprocess per image
Results saved to /kaggle/working/runs/detect/train